src/lib/math/bigint/divide.cpp

```diff
@@ -272,27 +277,30 @@ void vartime_divide(const BigInt& x, const BigInt& y_arg, BigInt& q_out, BigInt&
       const word x_j1 = r.word_at(j - 1);
       const word x_j2 = r.word_at(j - 2);
 
-      word qjt = bigint_divop_vartime(x_j0, x_j1, y_t0);
-
-      qjt = CT::Mask<word>::is_equal(x_j0, y_t0).select(WordInfo<word>::max, qjt);
+      word qjt = (x_j0 == y_t0) ? WordInfo<word>::max : bigint_divop_vartime(x_j0, x_j1, y_t0);
 
       // Per HAC 14.23, this operation is required at most twice
-      qjt -= division_check(qjt, y_t0, y_t1, x_j0, x_j1, x_j2);
-      qjt -= division_check(qjt, y_t0, y_t1, x_j0, x_j1, x_j2);
-      BOTAN_DEBUG_ASSERT(division_check(qjt, y_t0, y_t1, x_j0, x_j1, x_j2) == false);
+      while(division_check_vartime(qjt, y_t0, y_t1, x_j0, x_j1, x_j2)) {
+         qjt--;
+      }
```

Comment:
Should we guard this loop with an upper bound then, if it is meant to run at most twice? 🤔

Reply:
The code is a bit cleaner without the upper bound but I guess it might prevent a miscompilation elsewhere from turning the division into an ~infinite loop.